File: src/main/java/land/leets/domain/interview/usecase/UpdateInterview.java
package land.leets.domain.interview.usecase;

import land.leets.domain.interview.domain.Interview;
import land.leets.domain.interview.presentation.dto.req.FixedInterviewRequest;
import land.leets.domain.interview.presentation.dto.req.InterviewAttendanceRequest;

public interface UpdateInterview {
    Interview byUser(InterviewAttendanceRequest request);

    Interview byAdmin(Long id, FixedInterviewRequest request);
}

File: src/main/kotlin/land/leets/domain/interview/usecase/UpdateInterview.kt
package land.leets.domain.interview.usecase

import land.leets.domain.interview.domain.Interview
import land.leets.domain.interview.presentation.dto.req.FixedInterviewRequest
import land.leets.domain.interview.presentation.dto.req.InterviewAttendanceRequest

interface UpdateInterview {
    fun byUser(request: InterviewAttendanceRequest): Interview
    fun byAdmin(id: Long, request: FixedInterviewRequest): Interview
}
